Rook fixes

cristiklein opened the pull request:
**What this PR does / why we need it**:

The rook-ceph-toolsbox image change and CrashLoops if created with the wrong Deployment. Pulling the latest version.

Also, the deploy script is not idempotent and may fail with:

```
error: 'owner' already has a value (operator), and --overwrite is false
```

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

N/A

**Public facing documentation PR** *(if applicable)*
<!-- https://github.com/elastisys/compliantkubernetes/pull/ -->

N/A

**Special notes for reviewer**:


**Checklist:**

- [X] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
  - [X] is completely transparent, will not impact the workload in any way.
  - [ ] requires running a migration script.
  - [ ] requires draining and/or replacing nodes.
  - [ ] will break the cluster.
        I.e. full cluster migration is required.

<!--
Here are the commit prefixes and comments on when to use them:
all: (things that touch on more than one of the areas below, or don't fit any of them)
apps: (changes to app installers e.g. rook)
docs: (documentation)
tests: (test related changes)
pipeline: (the pipeline)
config: (configuration, e.g. add/remove/change values under `config`)
bin: (changes to binaries or scripts)

Example commit prefix usage:

git commit -m "docs: Add instructions for how to do x"
-->

Xartos commented on rook/toolbox-deploy.yaml:
```diff
@@ -17,10 +18,15 @@ spec:
-          image: rook/ceph:v1.5.3
-          command: ["/tini"]
-          args: ["-g", "--", "/usr/local/bin/toolbox.sh"]
+          image: rook/ceph:master
```
Isn't `master` basically same as `latest`? I wouldn't like that

cristiklein replied:
I was as surprised as you are, but seems like Rook really insists on `master`:
- https://github.com/rook/rook/blob/master/deploy/examples/toolbox.yaml#L21
- https://hub.docker.com/r/rook/ceph/tags?page=1&name=latest
- https://hub.docker.com/r/rook/ceph/tags?page=1&name=master

Xartos commented on rook/toolbox-deploy.yaml:
```diff
@@ -17,10 +18,15 @@ spec:
-          command: ["/tini"]
-          args: ["-g", "--", "/usr/local/bin/toolbox.sh"]
+          image: rook/ceph:master
+          command: ["/bin/bash"]
```
Any reason to this change?

cristiklein replied:
CrashLoop. :smile: The newer images do not include `/tini`. We could use an older version of `rook/ceph` in the toolbox, but since this is not on the "critical path" rather it's for debugging, I think that giving the operator the latest and greatest image is nicer.

Xartos commented on rook/toolbox-deploy.yaml:
```diff
@@ -2,6 +2,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: rook-ceph-tools
+  namespace: rook-ceph # namespace:cluster
```
```suggestion
  namespace: rook-ceph
```

cristiklein replied:
I simply copied the file from here: https://github.com/rook/rook/blob/master/deploy/examples/toolbox.yaml#L5

Unsure why they added the comment there, but since I didn't change anything else, it felt weird to change only this.

Xartos commented on rook/toolbox-deploy.yaml:
```diff
@@ -37,10 +43,6 @@ spec:
               name: ceph-config
             - name: mon-endpoint-volume
               mountPath: /etc/rook
-          resources:
-            requests:
-             cpu: "5m"
-             memory: "10Mi"
```
Why remove resource request? This doesn't follow our best practice. I would rather add limits to it as well.

cristiklein replied:
Okay, so you argue that we need to depart a bit from the official toolbox manifest, right?

Or should we upstream the `resources`?

Xartos replied:
But otherwise I don't see an issue with deviating in this case since this is just a single manifest tool. So I don't see any burden to manage these small changes.

cristiklein commented:
@Xartos I addressed all your comments, specifically:

- made container image stable
- added resource requests
- removed comment after namespace

I also added a comment to clearly document origin of file and what changes we made. PTAL.

Xartos reviewed (approved):
LGTM!
